### PR TITLE
Add map view functionality

### LINKED ISSUE #4: Map
# Goal

Create the map screen, and all attached functionality.

# Solution

[`react-native-maps`](https://github.com/react-native-maps/react-native-maps) provides a native map component.

# Tasks

### PR DESCRIPTION
**Pull request by charleswong739**
## Problem

Add map functionality. Closes #4.

## Description of Changes

- Expands `facilities.ts` database
- Builds Android to load Google API key (`.env` file not committed)

image

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -17,15 +17,26 @@ import { LoginContext } from './contexts/AppContext';
...
 import type { ImageSourcePropType } from 'react-native'
 import type { BottomTabNavigationOptions } from '@react-navigation/bottom-tabs';
 
 const Stack = createNativeStackNavigator();
 const Tab = createBottomTabNavigator();
+const HomeStack = createNativeStackNavigator();
 
 SplashScreen.preventAutoHideAsync();
 
+function HomeStackContainer() {
```
Comment (charleswong739) [inline App.tsx:31]: Created new navigation stack for the flows branching from the home screen.

File: App.tsx [hunk 2]
```diff
@@ -122,7 +135,7 @@ function App() {
 
   return (
     <NavigationContainer onReady={onLayoutRootView}>
-      <StatusBar backgroundColor={EStyleSheet.value('$pageBackgroundColor')} barStyle='dark-content' />
+      <StatusBar backgroundColor={'transparent'} translucent={true} barStyle='dark-content' />
```
Comment (charleswong739) [inline App.tsx:138]: Changed to transparent status bar so icons would display well on map screen.

File: android/app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    compileSdkVersion rootProject.ext.compileSdkVersion
+
+    namespace 'com.anonymous.STG'
+    defaultConfig {
+        applicationId 'com.anonymous.STG'
+        minSdkVersion rootProject.ext.minSdkVersion
+        targetSdkVersion rootProject.ext.targetSdkVersion
+        versionCode 1
+        versionName "1.0.0"
+        manifestPlaceholders = [GOOGLE_API_KEY: '$process.env.GOOGLE_API_KEY']
```
Comment (charleswong739) [inline android/app/build.gradle:91]: Read API key from `.env` file

File: android/app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        <action android:name="android.intent.action.VIEW"/>
+        <category android:name="android.intent.category.DEFAULT"/>
+        <category android:name="android.intent.category.BROWSABLE"/>
+        <data android:scheme="com.anonymous.STG"/>
+      </intent-filter>
+    </activity>
+    <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" android:exported="false"/>
+    <meta-data
+      android:name="com.google.android.geo.API_KEY"
+      android:value="${GOOGLE_API_KEY}" />
```
Comment (charleswong739) [inline android/app/src/main/AndroidManifest.xml:34]: Use Google API key.

File: screens/HomeScreen.tsx [hunk 5]
```diff
@@ -149,15 +149,15 @@ function HomeScreen({ navigation }) {
         </View>
       </View>
       <View style={styles.viewContainer}>
-        <Pressable style={styles.viewThumbnailContainer}>
+        <Pressable style={styles.viewThumbnailContainer} onPress={() => { console.log("Opening map view"); navigation.navigate("MapScreen"); }}>
           <Image source={require('../assets/map_thumbnail.png')} style={styles.viewThumbnail} />
-          <ShadowButton shadow={false} style={styles.viewButton}>
+          <ShadowButton shadow={false} style={styles.viewButton} disabled={true}>
```
Comment (charleswong739) [inline screens/HomeScreen.tsx:154]: Enabling the button will cause it to block hits for the underlying Pressable.

**Comment by annaa-cho** [comment]
Really happy w the pin sizing here, I don't think it's too big!
Suggestions for changes:
* let's move the popup a little more down, want to have about equal margins on all sides
* should we increase the shadow for the four options on the right?
* the progress bar doesn't really reflect the percentage—can we change this or is that not worth the work
* let's move the expand button bit more to the right to make it equidistant
* increased line spacing for the body text, and then we can adjust the spacing accordingly (move the title up + have less of an award space on the top/bottom of the 'open now' pill
* reduce the width of the 'open now' container
* the 'open now'/'closes in 3h' should have an equal amount of space on the top and bottom

**Comment by annaa-cho** [comment]
can we move the popup a tiny bit down, the donation centre info (all of it) a little bit up, and have equal spacing between the image, the progress bar, the progress bar text, and the end of the popup container?

**Review by annaa-cho — APPROVED** [review]

**Pull request merged**